src/crypto/secp256k1_provider/secp256k1_provider_impl.cpp

```diff
@@ -3,129 +3,92 @@
     OUTCOME_TRY(digest, sha256(message));
-    OUTCOME_TRY(public_key, bytesToPublicKey(key));
-    OUTCOME_TRY(result, VerifyEcSignature(digest, signature, public_key));
-    return result;
-  }
-
-  outcome::result<std::shared_ptr<EC_KEY>>
-  Secp256k1ProviderImpl::bytesToPrivateKey(const PrivateKey &input) {
-    std::shared_ptr<EC_KEY> key{EC_KEY_new_by_curve_name(NID_secp256k1),
-                                EC_KEY_free};
-    if (key == nullptr) {
-      return KeyGeneratorError::INTERNAL_ERROR;
-    }
-    std::unique_ptr<BIGNUM, void (*)(BIGNUM *)> key_bignum{
-        BN_bin2bn(input.data(), static_cast<int>(input.size()), nullptr),
-        BN_free};
-    if (key_bignum == nullptr) {
-      return KeyGeneratorError::INTERNAL_ERROR;
-    }
-    if (EC_KEY_set_private_key(key.get(), key_bignum.get()) != 1) {
-      return KeyGeneratorError::INTERNAL_ERROR;
-    }
-    std::unique_ptr<EC_POINT, void (*)(EC_POINT *)> point{
-        EC_POINT_new(EC_KEY_get0_group(key.get())), EC_POINT_free};
-    if (point == nullptr) {
-      return KeyGeneratorError::INTERNAL_ERROR;
-    }
-    EC_KEY_set_conv_form(key.get(), POINT_CONVERSION_COMPRESSED);
-    if (EC_POINT_mul(EC_KEY_get0_group(key.get()),
-                     point.get(),
-                     key_bignum.get(),
-                     nullptr,
-                     nullptr,
-                     nullptr)
-        != 1) {
-      return KeyGeneratorError::INTERNAL_ERROR;
-    }
-    if (EC_KEY_set_public_key(key.get(), point.get()) != 1) {
-      return KeyGeneratorError::INTERNAL_ERROR;
-    }
-    if (EC_KEY_check_key(key.get()) != 1) {
-      return KeyValidatorError::INVALID_PRIVATE_KEY;
-    }
-    return key;
-  }
-
-  outcome::result<std::shared_ptr<EC_KEY>>
-  Secp256k1ProviderImpl::bytesToPublicKey(const PublicKey &input) {
-    std::shared_ptr<EC_KEY> key{EC_KEY_new_by_curve_name(NID_secp256k1),
-                                EC_KEY_free};
-    if (key == nullptr) {
-      return KeyGeneratorError::INTERNAL_ERROR;
-    }
-    const uint8_t *public_key_ptr = input.data();
-    EC_KEY *key_ptr = key.get();
-    key_ptr = o2i_ECPublicKey(
-        &key_ptr, &public_key_ptr, static_cast<long>(input.size()));
-    if (key_ptr == nullptr) {
-      return KeyGeneratorError::INTERNAL_ERROR;
+    secp256k1_pubkey ffi_pub;
+    if (secp256k1_ec_pubkey_parse(ctx_.get(), &ffi_pub, key.data(), key.size())
+        == 0) {
+      return CryptoProviderError::SIGNATURE_VERIFICATION_FAILED;
     }
-    if (EC_KEY_check_key(key.get()) != 1) {
-      return KeyValidatorError::INVALID_PUBLIC_KEY;
+    secp256k1_ecdsa_signature ffi_sig;
+    if (secp256k1_ecdsa_signature_parse_der(
+            ctx_.get(), &ffi_sig, signature.data(), signature.size())
+        == 0) {
+      return CryptoProviderError::SIGNATURE_VERIFICATION_FAILED;
     }
-    return key;
+    return secp256k1_ecdsa_verify(ctx_.get(), &ffi_sig, digest.data(), &ffi_pub)
```

Comment:
This is perfectly fine and no need to change. Personally, I prefer splitting the line to assign a value of function to a temp variable for making easier possible debugging:
```suggestion
    auto verify_res = secp256k1_ecdsa_verify(ctx_.get(), &ffi_sig, digest.data(), &ffi_pub);
    return verify_res == 1;
```